### DISCUSSION TITLE
problem to find first feasible solution

### DISCUSSION THREAD

**Discussion by gregy4** [google/or-tools#3870] [CP-SAT questions]
Hello,
  I found a problem where cp-sat solver can't find feasible solution in hours with 16 search workers. For same problem constraint solver (choco solver) found first solution in a minute, the solution works as a hint for cp-sat solver.

  Since the problem contains about 3000 tasks with worker assignments represented by optional intervals (40 workers), total number of intervals is under 74 thousands with noOverlap constraint for every worker.

  Do I reached some limit where cp-sat solver doesn't work well ?

  Proto file is attached.
  
[model.zip](https://github.com/google/or-tools/files/12194347/model.zip)

Thanks,
Jan

**Comment by lperron** [comment]
Indeed, we know we have not good first solution heuristics for large scheduling problems.
Thanks for the proto.

**Reply by gregy4** [reply]
Thanks for info.
This problem can be solved with capacities instead individual workers, a group of same workers represented by a superworker with a capacity equal to a number of workers in a group. For complete solution, superworkers should be converted back to individual workers. For this problem optimal solution with superworkers can be found in minutes.
Unfortunately sometimes I have to work directly with individual workers.

**Comment by lperron** [comment]
But we have lower bounds :-)

```
Starting search at 82.82s with 64 workers.
14 full problem subsolvers: [core, default_lp, fixed, lb_tree_search, max_lp, no_lp, objective_lb_search, objective_shaving_search, probing, probing_max_lp, pseudo_costs, quick_restart, quick_restart_no_lp, reduced_costs]
48 first solution subsolvers: [jump, jump_decay_no_rst, jump_decay_perturb(11), jump_decay_rnd_on_rst(10), jump_no_rst, random(12), random_quick_restart(12)]
12 incomplete subsolvers: [feasibility_pump, graph_arc_lns, graph_cst_lns, graph_dec_lns, graph_var_lns, rins/rens, rnd_cst_lns, rnd_var_lns, scheduling_intervals_lns, scheduling_precedences_lns, scheduling_resource_windows_lns, scheduling_time_window_lns]
3 helper subsolvers: [neighborhood_helper, synchronization_agent, update_gap_integral]
#Bound  84.38s best:inf   next:[1,40]     objective_shaving_search #vars=216544 #csts=224388
#Bound 158.01s best:inf   next:[2,40]     objective_shaving_search #vars=216544 #csts=224388
#Bound 213.30s best:inf   next:[3,40]     objective_shaving_search #vars=216544 #csts=224388
#Bound 281.31s best:inf   next:[4,40]     objective_shaving_search #vars=216544 #csts=224388
#Bound 346.88s best:inf   next:[5,40]     objective_shaving_search #vars=216544 #csts=224388
#Bound 420.80s best:inf   next:[6,40]     objective_shaving_search #vars=216544 #csts=224388
#Bound 502.47s best:inf   next:[7,40]     objective_shaving_search #vars=216544 #csts=224388
#Bound 558.58s best:inf   next:[8,40]     objective_shaving_search #vars=216544 #csts=224388
#Bound 566.16s best:inf   next:[13,40]    lb_tree_search initial_propagation
```
